students/mia/users/views.py

```diff
@@ -1,3 +1,36 @@
 from django.shortcuts import render
 
 # Create your views here.
+import json
+import re 
+
+from django.http import JsonResponse
+from django.views import View
+
+from .models import User
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            input_data = json.loads(request.body)
+            
+            # 기존에 존재하는 이메일이 중복되면 중복 에러 메세지 반환
+            if User.objects.filter(email = input_data['email']).exists():
+                return JsonResponse({"message":"THE_USER_EMAIL_ALREADY_EXISTS"}, status=400)
+            # 이메일에 @ 또는 .이 없으면 에러 반환 
+            if re.match(r"^[a-zA-Z0-9+-_.]+@/[a-zA-Z0-9-.]+\.[a-zA-Z0-9-]$", input_data['email'])== None:
+                return JsonResponse({"message":"INVALID_EMAIL_--_NEEDS_@_AND_."}, status=400)
+            # 비밀번호는 8자리 이상, 문자 숫자, 특수문자의 복합 -- 그러지 않을 경우, 에러 반환
+            if re.match(r"^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$", input_data["password"])==None:
+                return JsonResponse({"message":"PASSWORD_REQUIRES_A_COMBINATION_OF_MIMIMUM_EIGHT_LETTERS,NUMBERS,AND_SPECIAL_SYMBOLS"}, status=400)
```

Comment:
파이썬 PEP 8 프로그래밍 권고에는 비교문 관련해서 어떤 형식을 사용하라고 안내되어 있습니다.

- True/False를 비교할 때는 if문에 변수만 사용하도록 한다.
- 동등성(단순히 값만 비교하는 경우)을 검증할 때는 == 혹은 != 같은 연산자를 사용하도록 한다.
- 동일성(객체의 고유 메모리 주소로 비교하는 경우)을 검증할 때는 is나 is not을 사용한다.

```suggestion
            if not re.match(r"^[a-zA-Z0-9+-_.]+@/[a-zA-Z0-9-.]+\.[a-zA-Z0-9-]$", input_data['email']):
                return JsonResponse({"message":"INVALID_EMAIL_--_NEEDS_@_AND_."}, status=400)
            # 비밀번호는 8자리 이상, 문자 숫자, 특수문자의 복합 -- 그러지 않을 경우, 에러 반환
            if not re.match(r"^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$", input_data["password"]):
                return JsonResponse({"message":"PASSWORD_REQUIRES_A_COMBINATION_OF_MIMIMUM_EIGHT_LETTERS,NUMBERS,AND_SPECIAL_SYMBOLS"}, status=400)
```